crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -4,20 +4,24 @@
 //! we define the way to calculate the protocol fee based on the configuration
 //! parameters.
 
-use crate::{
-    boundary::{self},
-    domain,
+use {
+    crate::{
+        arguments,
+        boundary::{self},
+        domain,
+    },
+    primitive_types::U256,
 };
 
 /// Constructs fee policies based on the current configuration.
 #[derive(Debug)]
 pub struct ProtocolFee {
-    policy: Policy,
+    policy: arguments::FeePolicy,
```

Comment:
This should probably continue to take a `domain::FeePolicy` to not tie the `domain` type to the representation we need to have for `CLI` parsing.

Reply:
I mean that usually the idea is to make the `domain` directories the kings. So for this example it would mean that the `arguments` code has to bend over backwards to produce a `domain::FeePolicy` type instead of making the `domain` use a type that is not 100% optimized for the `domain`.
If I'm reading the code correctly we'd now use a `arguments::FeePolicy` type which is only used to tell to allow users to pass that data via CLI but might not be the best choice for the `domain`.
Very often we use nearly identical types for different directories and converting between those types then looks like unnecessary boilerplate code. But for example if we were to change how arguments get passed into the program down the line this separation allows us to change the parsing without affecting the domain at all.